donkeycar/parts/tub_v2.py

```diff
@@ -105,21 +110,44 @@ class TubWriter(object):
     """
     def __init__(self, base_path, inputs=[], types=[], metadata=[],
                  max_catalog_len=1000):
-        self.tub = Tub(base_path, inputs, types, metadata, max_catalog_len)
+        self.tub = Tub(base_path, inputs + ['session_id'], types + ['str'],
```

Comment:
Lets take the inputs and types and store them in a `dict` in the beginning. That way we only need to manipulate the dict going forward.

Reply:
Makes sense.

Reply:
No, i meant the caller should include `session` and `str` in `inputs` and `types`. I feel like we are muddling the responsibilities a bit. The `run` method's responsibility should provide the `session_id` and not the other way around. The `Tub` should **not** have to discover the `session_id`. 

I think this will become easier when you store the `Tub` specific metadata into its own `manifest` file. See my comment below.